### ISSUE TITLE
Feature Request: Implement Save State Functionality in Xenia Canary

### ISSUE THREAD

**Issue by 7ROBE** [xenia-canary/xenia-canary#349]
### **Summary:**
I would like to request the implementation of a save state feature in Xenia Canary, similar to the one available in PCSX2 and RPCS3. Save states are an essential feature for many users, offering convenience and flexibility in gameplay.

### **Description:**
Save states allow users to save the exact state of an emulator at any point during gameplay and reload that state later. This feature is incredibly useful for testing, debugging, and casual playthroughs, as it allows users to:

1. **Save and Resume:** Save the game at any point, even if the in-game save system is unavailable or inconvenient, and resume from the exact point later.
2. **Experimentation:** Test various strategies or actions in games by reloading to a specific point without needing to replay large sections.
3. **Quick Saving:** Easily save progress in challenging sections without relying solely on in-game checkpoints.

Given the advanced state of Xenia Canary, implementing save states would significantly enhance the user experience and bring it on par with other emulators like PCSX2 and RPCS3.

### **Expected Behavior:**
- The ability to create save states at any point during gameplay.
- A simple user interface to manage, load, and delete save states.
- Compatibility with all supported games, or at least a large majority, without causing crashes or corruption.

### **Additional Context:**
Save states have become a staple in modern emulation, and their absence in Xenia Canary limits the usability and appeal of the emulator. With this feature, users can enjoy a more seamless and customizable gaming experience.

Thank you for considering this feature request. I believe it would greatly benefit the Xenia Canary community and enhance the overall functionality of the emulator.

**Comment by BigBankBailey** [comment]
Yes we need this for when the games crash we don't have to wait forever again to play.

**Comment by 55TeMeR55** [comment]
+
